plugin.video.retrospect/resources/libs/retroconfig.py

```diff
@@ -0,0 +1,80 @@
+        #noinspection PyArgumentEqualDefault
+        __path = __path.decode('utf-8')
+
+    # get rootDir, addonsDir and profileDir
+    rootDir = __path.replace(";", "")                        # : The root directory where Retrospect resides.
+    addonDir = os.path.split(rootDir)[-1]                    # : The add-on directory of Kodi.
+    rootDir = os.path.join(rootDir, '')                      # : The root directory where Retrospect resides.
+
+    # determine the profile directory, where user data is stored.
+    if xbmc.getCondVisibility("system.platform.xbox"):
+        profileDir = os.path.join(xbmc.translatePath("special://profile/script_data/"), addonDir)
+        profileUri = os.path.join("special://profile/script_data/", addonDir)
+    else:
+        profileDir = os.path.join(xbmc.translatePath("special://profile/addon_data/"), addonDir)
+        profileUri = os.path.join("special://profile/addon_data/", addonDir)
+
+    # the XBMC libs return unicode info, so we need to convert this
+    if isinstance(profileDir, bytes):
+        profileDir = profileDir.decode('utf-8')
+    if isinstance(profileUri, bytes):
+        profileUri = profileUri.decode('utf-8')
+
+    cacheDir = os.path.join(profileDir, 'cache', '')         # : The cache directory.
+    favouriteDir = os.path.join(profileDir, 'favourites')    # : The favourites directory
+
+    appName = "Retrospect"                                   # : Name of the XOT application (could be overwritten from the addon.xml)
+    cacheValidTime = 7 * 24 * 3600                           # : Time the cache files are valid in seconds.
+
+    logLevel = 10                                            # : Minimum log level that is being logged. (from logger.py) Defaults to Debug
+    logFileNameAddon = "retrospect.log"                      # : Filename of the log file of the plugin
+
+    googleAnalyticsId = "UA-3902785-11"                      # : Google Analytics ID for statistics
+
+    # must be single quotes for build script
+    __addonXmlPath = os.path.join(rootDir, 'addon.xml')
+    __addonXmlcontents = xml.dom.minidom.parse(__addonXmlPath)
+    for addonentry in __addonXmlcontents.getElementsByTagName("addon"):
+        addonId = str(addonentry.getAttribute("id"))          # : The ID the addon has in Kodi (from addon.xml)
+        __version = addonentry.getAttribute("version")        # : The Version of the addon (from addon.xml) in text
+        version = Version(version=__version)                  # : The Version of the addon (from addon.xml)
+        #noinspection PyRedeclaration
+        appName = str(addonentry.getAttribute("name"))        # : The name from the addon (from addon.xml)
+
+    updateUrl = "https://api.bitbucket.org/2.0/repositories/basrieter/xbmc-online-tv/downloads/"
+
+    textureMode = "Cached"                                    # : The mode for the textures: Local, Remote or Cached
+    textureUrl = \
```

Comment:
...

Reply:
This is an configuration option to switch between, 

* Cached: locally cached textures
* Remote: remote textures (Kodi caches the online url's) 
* Local (included in the zip)

It basically uses a small factory pattern to determine how to use the textures. `Cached` is the fastest one for the user, so I use that for now.